pkg/networkpolicy/controller.go

```diff
@@ -173,6 +176,24 @@ func newController(client clientset.Interface,
 
 	// reduce memory usage only care about Labels and Status
 	trim := func(obj interface{}) (interface{}, error) {
+		if po, ok := obj.(*v1.Pod); ok {
+			ips := make([]v1.PodIP, 0, len(po.Status.PodIPs))
+			for _, i := range po.Status.PodIPs {
+				ips = append(ips, v1.PodIP{IP: intern(i.IP)})
+			}
+			return &v1.Pod{
```

Comment:
is not replacing in place cheaper?

Reply:
ah, apparently `SetTransform` is the exception to the normal "you can't modify objects from the informer cache" rule

Reply:
it runs before goes to the informer cache IIRC

Reply:
The reason I didn't replace in place (and made sure to copy every item entirely) is due to how Go will sometimes not GC a full struct if you hold a reference to something within the struct. https://stackoverflow.com/a/55018900

However, I doubt this applies here, or if it is it can be avoided in a simpler manner -- I was measuring things wrong so took an overkill approach. ill clean it up